mithril-client/src/services/snapshot.rs

```diff
@@ -260,21 +275,25 @@ impl SnapshotService for MithrilClientSnapshotService {
             })?;
 
         progress_bar.println("7/7 - Verifying the snapshot signature…")?;
-        let mut protocol_message = certificate.protocol_message.clone();
-        protocol_message.set_message_part(
-            ProtocolMessagePartKey::SnapshotDigest,
-            unpacked_snapshot_digest,
-        );
-        if protocol_message.compute_hash() != certificate.signed_message {
+        let expected_message = {
```

Comment:
This change seems unrelated to the PR's purpose ?

Reply:
Yes, I did a "boy scoot" on the file clean up before working on this issue. The bulk of the code implementing the issue is on the second commit "Make client append a empty 'clean' file after download succeed".